x-pack/platform/plugins/shared/alerting/server/application/rule/methods/get_rule_ids_with_gaps/get_rule_ids_with_gaps.ts

```diff
@@ -62,38 +83,42 @@ export async function getRuleIdsWithGaps(
       {
         filter,
         aggs: {
-          latest_gap_timestamp: {
-            max: {
-              field: '@timestamp',
-            },
-          },
-          unique_rule_ids: {
-            terms: {
-              field: 'rule.id',
-              size: 10000,
-            },
+          latest_gap_timestamp: { max: { field: '@timestamp' } },
+          by_rule: {
+            terms: { field: 'rule.id', size: 10000, order: { _key: 'asc' } },
+            aggs: COMMON_GAP_AGGREGATIONS,
           },
         },
       }
     );
 
-    interface UniqueRuleIdsAgg {
-      buckets: Array<{ key: string }>;
+    interface ByRuleBucket extends GapDurationBucket {
+      key: string;
     }
 
-    const uniqueRuleIdsAgg = aggs.aggregations?.unique_rule_ids as UniqueRuleIdsAgg;
-    const latestGapTimestampAgg = aggs.aggregations?.latest_gap_timestamp as { value: number };
-
-    const resultBuckets = uniqueRuleIdsAgg?.buckets ?? [];
+    const byRuleAgg = aggs.aggregations?.by_rule as { buckets: ByRuleBucket[] };
+    const buckets = byRuleAgg?.buckets ?? [];
 
-    const ruleIds = resultBuckets.map((bucket) => bucket.key) ?? [];
+    const ruleIds: string[] = [];
+    if (highestPriorityGapFillStatuses?.length ?? 0 > 0) {
+      for (const b of buckets) {
+        const sums = extractGapDurationSums(b);
+        const gapFillStatus = calculateHighestPriorityGapFillStatus(sums);
+        if (gapFillStatus && highestPriorityGapFillStatuses?.includes(gapFillStatus))
+          ruleIds.push(b.key);
+      }
+    } else {
+      for (const b of buckets) {
+        ruleIds.push(b.key);
+      }
+    }
 
+    const latestGapTimestampAgg = aggs.aggregations?.latest_gap_timestamp as { value: number };
```

Comment:
I'm just curious why using manually written interfaces like `{ value: number }` instead of importing aggregation types from `@elastic/elasticsearch/lib/api/types`? For example `AggregationsMaxAggregate` could be used here instead of `{ value: number }`.

I tried to apply `AggregationsMaxAggregate` type but it revealed a discrepancy between types in the result. It may mean we return values we aren't expecting to return.

<img width="898" height="309" alt="image" src="https://github.com/user-attachments/assets/a4009777-6206-4221-9b3f-6894628a8516" />